components/suggestion/search_date_suggestion/index.ts

```diff
@@ -34,7 +34,7 @@ function makeMapStateToProps() {
         }
 
         return {
-            currentDate,
+            currentDate: currentDate as Date,
```

Comment:
Similarly, `SearchDateSuggestion` can handle when `currentDate` is undefined, so I think this prop should be optional as well